### PR TITLE
Update version of Gradle and libraries

### PR DESCRIPTION
**Pull request by krysthianlessa**
## Description

This PR aims to upgrade the gradle version, and also the libraries used in the project. The table below shows the upgrades:

| Artifact | Old version | New version |
| --------- | -------------- | ---------------- |
| Gradle |  `6.7.1` | `7.5` |
| Kotlin |  `1.5.31` | `1.9.10` |
| Android SDK |  `30` | `33` |
| android-gradle-plugin |  `4.2.2` | `7.4.2` |
| appcompat |  `1.3.1` | `1.6.1` |
| constraintlayout |  `2.1.2` | `2.1.4` |
| junit |  `1.1.3` | `1.1.5` |
| espresso-core |  `3.4.0` | `3.5.1` |

### View Binding changes

In this new version of the Android SDK, Kotlin Android Extensions are deprecated, which means that using Kotlin synthetics for view binding is no longer supported. How the project used Kotlin synthetics for view binding, this [guide](https://developer.android.com/topic/libraries/view-binding/migration) was used to migrate to Jetpack view binding.

For this it was necessary: 

- Set the `viewBinding` build option to `true` in the module-level `build.gradle` file:

```
android {
    ...
    buildFeatures {
        viewBinding = true
    }
}
```
- Inflate the generated binding class instance for the activity to use (guide [here](https://developer.android.com/topic/libraries/view-binding#activities)):

To set up an instance of the binding class for use with an activity, perform the following steps in the activity's `onCreate() `method:

1. Call the static inflate() method included in the generated binding class. This creates an instance of the binding class for the activity to use.
2. Get a reference to the root view by either calling the `getRoot()` method or using Kotlin property syntax.
3. Pass the root view to `setContentView()` to make it the active view on the screen.

These steps resulted in the following code:

```
...
private lateinit var binding: ActivityMainBinding

override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
}
...
```

### CODE REVIEW

**Comment by Marinofull** [comment]
The CodeQL failure has no relations with this PR, see https://github.com/oxeanbits/parsecs-android/actions
image

File: demo/src/main/java/br/com/nvsistemas/parsec/demo/MainActivity.kt [hunk 1]
```diff
@@ -3,27 +3,24 @@ package br.com.nvsistemas.parsec.demo
...
-        show_result_btn.setOnClickListener {
-            result.text = eval(expression.text.toString())
+        binding.showResultBtn.setOnClickListener {
+            binding.result.text = eval(binding.expression.text.toString())
         }
 
-        show_result_btn2.setOnClickListener {
-            result2.text = eval(expression2.text.split(",").map { it.trim() })
+        binding.showResultBtn2.setOnClickListener {
+            binding.result2.text = eval(binding.expression2.text.split(",").map { it.trim() })
```
Comment (Marinofull) [inline demo/src/main/java/br/com/nvsistemas/parsec/demo/MainActivity.kt:23]: @rodiguif can you review if we can proceed to merge it all?

File: demo/src/main/java/br/com/nvsistemas/parsec/demo/MainActivity.kt [hunk 2]
```diff
@@ -3,27 +3,24 @@ package br.com.nvsistemas.parsec.demo
...
 class MainActivity : AppCompatActivity() {
 
     private val parsec = Parsec()
+    private lateinit var binding: ActivityMainBinding
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
```
Comment (rodiguif) [inline demo/src/main/java/br/com/nvsistemas/parsec/demo/MainActivity.kt:16]: Interesting :thinking: 

Could you explain better these changes in the description?

File: demo/build.gradle [hunk 3]
```diff
@@ -24,18 +23,22 @@ android {
         }
     }
 
+    viewBinding {
+        enabled = true
+    }
```
Comment (rodiguif) [inline demo/build.gradle:28]: Same here.

**Review by rodiguif — CHANGES_REQUESTED** [review]
Worked well! Also tested in DPMS. :no_smoking:  :heavy_check_mark:
Good job, dude!

We just need to better document some of these changes:

**Pull request merged**